File: serial_hpmprof/sub.c
/* Copyright IBM Corporation, 2019
 * author : Bob Walkup
 */

double sub(double x)
{
    int iter, maxiter;
    int i, npts;
    double val = 0.0;

    npts = 1000000;
    maxiter = 1000;

    for (iter=0; iter<maxiter; iter++) {
      for (i=0; i<npts; i++) val += x * (double) (i%10);
    }

    return val;
}

File: mpi_examples/sub.f90
! Copyright IBM Corporation, 2019
! author : Bob Walkup
!

subroutine sub()
  implicit none
  include 'mpif.h'
  integer myrank, val, info
  call mpi_comm_rank(mpi_comm_world, myrank, info)
  if (myrank == 0) then
    val = 1
  else
    val = -1
  end if
  call mpi_bcast(val, 1, mpi_integer, 0, mpi_comm_world, info)
end
